File: graniteApiClient.js
const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;

// NOTE: you must manually enter your API_KEY below using information retrieved from your IBM Cloud account (https://eu-gb.dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/ml-authentication.html?context=wx)
const API_KEY = "<your API key>";

function getToken(errorCallback, loadCallback) {
	const req = new XMLHttpRequest();
	req.addEventListener("load", loadCallback);
	req.addEventListener("error", errorCallback);
	req.open("POST", "https://iam.cloud.ibm.com/identity/token");
	req.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
	req.setRequestHeader("Accept", "application/json");
	req.send("grant_type=urn:ibm:params:oauth:grant-type:apikey&apikey=" + API_KEY);
}

function apiPost(scoring_url, token, payload, loadCallback, errorCallback){
	const oReq = new XMLHttpRequest();
	oReq.addEventListener("load", loadCallback);
	oReq.addEventListener("error", errorCallback);
	oReq.open("POST", scoring_url);
	oReq.setRequestHeader("Accept", "application/json");
	oReq.setRequestHeader("Authorization", "Bearer " + token);
	oReq.setRequestHeader("Content-Type", "application/json;charset=UTF-8");
	oReq.send(payload);
}

getToken((err) => console.log("An error occurred submitting the request."), () => {
	let tokenResponse;
	try {
		tokenResponse = JSON.parse(this.responseText);
	} catch(ex) {
		// TODO: handle parsing exception
	}
	// NOTE:  manually define and pass the array(s) of values to be scored in the next line
	const payload = `{"messages":[{"content":"","role":""}]}`;

	const scoring_url = "https://eu-gb.ml.cloud.ibm.com/ml/v4/deployments/3e0b061f-1365-4329-8a43-bda3631cd503/ai_service_stream?version=2021-05-01";
	apiPost(scoring_url, tokenResponse.access_token, payload, function (resp) {
		let parsedPostResponse;
		try {
			parsedPostResponse = JSON.parse(this.responseText);
		} catch (ex) {
			// TODO: handle parsing exception
		}
		console.log("Scoring response");
		console.log(parsedPostResponse);
	}, function (error) {
		console.log(error);
	});
});

File: GraniteApiClient.scala
import scalaj.http.{Http, HttpOptions}
import scala.util.{Success, Failure}
import java.util.Base64
import java.nio.charset.StandardCharsets
import play.api.libs.json._

// NOTE: you must manually set API_KEY below using information retrieved from your IBM Cloud account (https://eu-gb.dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/ml-authentication.html?context=wx)

val API_KEY = "<your API key>"

// Get IAM service token
val iam_url = "https://iam.cloud.ibm.com/identity/token"
val iam_response = Http(iam_url).header("Content-Type", "application/x-www-form-urlencoded").header("Accept",
 "application/json").postForm(Seq("grant_type" -> "urn:ibm:params:oauth:grant-type:apikey",
  "apikey" -> API_KEY)).asString
val iamtoken_json: JsValue = Json.parse(iam_response.body)

val iamtoken = (iamtoken_json \ "access_token").asOpt[String] match {
	case Some(x) => x
	case None => ""
}

// TODO:  manually define and pass list of values to be scored
val payload_scoring: JsValue = Json.parse("""{"messages":[{"content":"","role":""}]}""")

val scoring_url = "https://eu-gb.ml.cloud.ibm.com/ml/v4/deployments/3e0b061f-1365-4329-8a43-bda3631cd503/ai_service_stream?version=2021-05-01"

val response_scoring = Http(scoring_url).postData(payload_scoring).header("Content-Type",
 "application/json").header("Authorization", "Bearer " + iamtoken).option(HttpOptions.
	method("POST")).option(HttpOptions.connTimeout(10000)).option(HttpOptions.readTimeout(50000)).asString
println("scoring response")
println(response_scoring)
